src/createReducedStream.ts

```diff
@@ -0,0 +1,54 @@
+import { BehaviorSubject, Observable, Subscription, Subject } from 'rxjs';
+import { ActionStream } from './types/helpers';
+import { RegisteredReducer, combineReducers } from './reducer';
+import { defaultErrorSubject } from './internal/defaultErrorSubject';
+import { markName } from './internal/markers';
+import { tag } from 'rxjs-spy/operators';
+
+export class ReducedStream<State> extends BehaviorSubject<State> {
```

Comment:
From your PR description:
> The current implementation in this PR extends BehaviorSubject and subscribes the stream when created. This is not ideal, as next, error and complete are available as methods on the objects, though they should never be used. It also means the state can only be exposed through the property name value, which is very non-descriptive.

I think it's best if we can conclude on whether it's good or bad that `ReducedStream` is a behavior subject.

Perhaps it makes sense to rather implement our own stream which doesn't have a `.next()` method? I think it still makes sense to have `.complete()` so it could work with `takeUntil` rxjs operators.

I think the property name `value` is fine, since it's consistent with RxJS subjects.

Reply:
This is the main reason this PR is WIP :)

I had some trouble formulating the logic when extending `Observable`, so I opted for extending `BehaviorSubject` just to get started. I'll look into `Observable` again.

Reply:
👍